### ISSUE TITLE
Apply transformations from the Cloudiary URL

### ISSUE THREAD

**Issue by danielolaviobr** [colbyfayock/next-cloudinary#9]
### TLDR;
As discussed in #2 the Cloudiary URL may contain transformations that apply to the image. We would like to be able to apply the modification present in the url to the image component.

## Situation
The Cloudiary URL may have transformations in it, that are applied to the image it links to. The logic to be able to handle  a Cloudiary URL in the `CldImage` component is being implemented in #8. But as discussed in #2 it would be too complex for this first implementation to handle these transformations.

An url with transformations applied to it looks like the following:
```
https://res.cloudinary.com/colbycloud-next-cloudinary/image/upload/w_100,h_100/v1/images/turtle
```
And returns an image like this:
![transformed image](https://user-images.githubusercontent.com/64712584/193569161-e2ec73fb-3f2b-4a72-a783-a9a985acad1f.png)

While a url without transformation looks like this:
```
https://res.cloudinary.com/colbycloud-next-cloudinary/image/upload/v1/images/turtle
```
And returns an image like this:
![normal image](https://user-images.githubusercontent.com/64712584/193569338-24163a72-af08-4c03-bacc-5a8f801c1edc.png)

## Implementation
The goal would be able to parse the url and retrieve the transformation part `w_100,h_100`, then parse this string into the component transformations and apply them. A possible extra feature, to avoid overriding transformations would be to have a `useUrlTransformations` prop in the component that would let us know which transformations to use, the ones in the url or the ones in the props of the component

**Comment by colbyfayock** [comment]
`useUrlTransformations` or maybe `preserveTransformations`, but agree it should be controllable!

**Comment by colbyfayock** [comment]
@danielolaviobr do you plan on working on this one or should i open it up to others? thanks

**Comment by danielolaviobr** [comment]
I've being trying to find time to work on it, but since I came back from vacation I couldn't. You can open it, I'll try to help as I can in the issue

**Comment by colbyfayock** [comment]
sounds great thanks @danielolaviobr

**Comment by Sambhav250703** [comment]
Can you guide me pls basically it will my first contribution in node js

**Comment by colbyfayock** [comment]
hey @Sambhav250703 do you mean that you dont have a lot of experience with Node.js? if that's the case, i might recommend maybe you can start off instead by tackling one of the Test Issues that are open as a way to get your feet wet, this one likely will be complicated

**Comment by Sambhav250703** [comment]
Sure thanks for guidance

**Comment by colbyfayock** [comment]
yes for sure @agbanusi !

**Comment by colbyfayock** [comment]
@agbanusi do you plan on working on this issue still?

**Comment by agbanusi** [comment]
Sorry about this, I already worked on it but didn't have the chance to test the changes which I planned to do this weekend. This new PR is what I have so far and what's left for me is testing and documentation and please help me review it so I could make any required changes needed
